docs: add docs how to patch in kubernetes (#5230)
jina-ai/jina: docs/how-to/kubernetes.md

print(f"Matched documents: {len(matches)}")
```

## Update your Executor in Kubernetes

In Kubernetes, you can update your Executors by patching the deployment corresponding to your Executor.

For instance, in the example above, we may be interested at some point on changing the `batch_size` parameter of the CLIPEncoder.

> In order to do so, you need to change the content of the Deployment inside the `executor.yml` dumped by `.to_kubernetes_yaml`.

Reviewer: ```suggestion
To do this, change the content of the Deployment inside the `executor.yml` dumped by `.to_kubernetes_yaml`.
```

> You need to update the `arguments` passed to the `container` inside the `Deployment`.

Reviewer: ```suggestion
You need to update the `arguments` passed to the `container` inside the `Deployment` as follows:
```

print(f"Matched documents: {len(matches)}")
```

## Update your Executor in Kubernetes

> In Kubernetes, you can update your Executors by patching the deployment corresponding to your Executor.

Reviewer 2: "Deployment" or "deployment"? I think either is fine because it is a k8s term, but not only; but we should capitalize it consistently (here it isn't, below it is).

> For instance, in the example above, we may want to change the `batch_size` parameter of the CLIPEncoder.

Reviewer 2: ```suggestion
For instance, in the example above, we may want to change set a `batch_size` parameter for the CLIPEncoder.
```

To do this, change the content of the Deployment inside the `executor.yml` dumped by `.to_kubernetes_yaml`.

> You need to update the `arguments` passed to the `container` inside the `Deployment` as follows:

Reviewer 2: ```suggestion
You need to add `--uses_with` and pass the batch size argument to it. This will be passed to the container inside the Deployment:
```

```yaml
    spec:
      containers:
      - args:
        - executor
        - --name
        - encoder
        - --k8s-namespace
        - custom-namespace
        - --uses
        - config.yml
        - --port
        - '8080'
        - --uses-metas
        - '{}'
        - --uses-with
        - '{"batch_size": 64}'
        - --native
        command:
        - jina
```

After doing so, you can run:

```shell script

> kubectl patch deployment encoder --patch-file ./k8s_flow/encoder/encoder.yml -n k8s-namespace

Reviewer 2: ```suggestion
kubectl patch deployment encoder --patch-file ./k8s_flow/encoder/encoder.yml -n custom-namespace
```

```

> This will make sure that the Pods are restarted with the new parameters, which will make the Executor to be restarted in a new docker container 

Reviewer 2: ```suggestion
This will make sure that the Pods are restarted with the new parameters, which will make the Executor to be restarted in a new Docker container 
```
